src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -184,6 +194,27 @@ public static bool IsDisposeImplementation([NotNullWhen(returnValue: true)] this
                 method.IsImplementationOfInterfaceMethod(null, iDisposable, "Dispose");
         }
 
+        /// <summary>
+        /// Checks if the given method implements "IAsyncDisposable.Dispose" or overrides an implementation of "IAsyncDisposable.Dispose".
+        /// </summary>
+        public static bool IsAsyncDisposeImplementation([NotNullWhen(returnValue: true)] this IMethodSymbol? method, [NotNullWhen(returnValue: true)] INamedTypeSymbol? iAsyncDisposable, INamedTypeSymbol? valueTaskType)
+        {
+            if (method == null)
+            {
+                return false;
+            }
+
+            if (method.IsOverride)
```

Comment:
I see that this is a duplicate of what's in place for `Dispose` but I am not sure to understand what's the point of doing the check on the overriden method. AFAIK the override method has the same signature, isn't it?

Reply:
@Evangelink I'm not sure about that. I just duplicated the logic that already exists. @mavasani do you know?